flow/probes/ebpf.go

```diff
@@ -261,45 +275,59 @@ func (p *EBPFProbesHandler) Stop() {
 	p.wg.Wait()
 }
 
-func LoadJumpMap(module *elf.Module) error {
-	var jmpTable []string = []string{"socket_network_layer"}
+func LoadJumpMap(module *ebpf.Collection) error {
+	var jmpTable []string = []string{"network_layer"}
 
-	jmpTableMap := module.Map("jmp_map")
+	jmpTableMap := module.Maps["jmp_map"]
 	if jmpTableMap == nil {
 		return fmt.Errorf("Map: jmp_map not found")
 	}
 	for i, sym := range jmpTable {
-		entry := module.SocketFilter(sym)
+		entry := module.Programs[sym]
 		if entry == nil {
 			return fmt.Errorf("Symbol %s not found", sym)
 		}
 
-		index := uint32(i)
-		fd := uint32(entry.Fd())
-		err := module.UpdateElement(jmpTableMap, unsafe.Pointer(&index), unsafe.Pointer(&fd), BPF_ANY)
+		fd := uint32(entry.FD())
+		err := jmpTableMap.Put(uint32(i), &fd)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (p *EBPFProbesHandler) loadModuleFromAsset(path string) (*elf.Module, error) {
+func (p *EBPFProbesHandler) loadModuleFromAsset(path string) (*ebpf.Collection, error) {
 	data, err := statics.Asset(path)
 	if err != nil {
 		return nil, fmt.Errorf("Unable to find eBPF elf binary in bindata")
 	}
 
-	module := elf.NewModuleFromReader(bytes.NewReader(data))
-	err = module.Load(nil)
+	collspec, err := ebpf.LoadCollectionSpecFromReader(bytes.NewReader(data))
+	if err != nil {
+		return nil, fmt.Errorf("Can't load %s: %v\n", path, err)
+	}
+
+	module, err := ebpf.NewCollection(collspec)
+	if err != nil {
+		return nil, fmt.Errorf("Can't create collection %s: %v\n", path, err)
+	}
 
 	if err == nil {
 		p.Ctx.Logger.Infof("Loaded eBPF module ", path)
 	}
 	return module, err
 }
 
-func (p *EBPFProbesHandler) loadModule() (*elf.Module, error) {
+func (p *EBPFProbesHandler) loadModule() (*ebpf.Collection, error) {
+	err := syscall.Setrlimit(8, &syscall.Rlimit{
```

Comment:
I assume the `8` here is for `RLIMIT_MEMLOCK`; I suggest adding a comment or using a C constant from `#include <sys/resource.h>`.

Reply:
Yes I forgot to add it, thanks